File: backend/core/src/main/java/com/careconnect/model/MailPiece.java
package com.careconnect.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.*;
import java.time.OffsetDateTime;

@Data @NoArgsConstructor @AllArgsConstructor @Builder
public class MailPiece {
    private String id;
    private String sender;

    @JsonProperty("summary")
    private String subject;

    @JsonProperty("imageDataUrl")
    private String thumbnailUrl;        // data: URL or https link

    private OffsetDateTime receivedAt;  // when the digest says it's from

    @JsonProperty("actions")
    private ActionLinks actionLinks;
}

File: frontend/lib/features/usps/domain/models/mail_piece.dart
import 'package:care_connect_app/features/usps/domain/models/action_links.dart';

class MailPiece {
  final String id;
  final String? sender, summary, imageDataUrl, dateIso;
  final ActionLinks actions;
  const MailPiece({required this.id, this.sender, this.summary, this.imageDataUrl, this.dateIso, required this.actions});
}